### ISSUE TITLE
syclcompat/memory/memory_management_test3.cpp failing on Gen12 Linux on unrelated changes

### ISSUE THREAD

**Issue by sarnex** [intel/llvm#14086]
### Describe the bug

https://github.com/intel/llvm/actions/runs/9405479179/job/25907121371?pr=14081

```
# |   sycl::aspect::fp64 not supported by the SYCL device.
# | void test_fill_async_q() [T = sycl::detail::half_impl::half]
# `-----------------------------
# RUN: at line 34
env ONEAPI_DEVICE_SELECTOR=opencl:gpu  /__w/llvm/llvm/build-e2e/syclcompat/memory/Output/memory_management_test3.cpp.tmp.out
# executed command: env ONEAPI_DEVICE_SELECTOR=opencl:gpu /__w/llvm/llvm/build-e2e/syclcompat/memory/Output/memory_management_test3.cpp.tmp.out
# .---command stdout------------
# | void test_free_memory()
# | void test_free_memory_q()
# | void test_wait_and_free_memory()
# `-----------------------------
# .---command stderr------------
# | terminate called after throwing an instance of 'sycl::_V1::runtime_error'
# |   what():  Native API failed. Native API returns: -30 (PI_ERROR_INVALID_VALUE) -30 (PI_ERROR_INVALID_VALUE)
# `-----------------------------
# error: command failed with exit status: -6
```

**Comment by AidanBeltonS** [comment]
Proposed fix with, #14080

**Comment by sarnex** [comment]
Fixed with https://github.com/intel/llvm/pull/14080
